libcxx/include/__chrono/time_zone.h

```diff
@@ -70,6 +72,30 @@ class _LIBCPP_AVAILABILITY_TZDB time_zone {
     return __get_info(chrono::time_point_cast<seconds>(__time));
   }
 
+  // Since the interface promisses throwing, don't add nodiscard.
```

Comment:
I don't think this comment is necessary -- we assume no `nodiscard` by default anyway, and I think anyone pausing to think whether this should be made `nodiscard` will easily come to the same conclusion.

Or if you want to keep the comment, I would reword to `We don't apply nodiscard here since this function throws on many inputs, so it could be used as a validation`.